### ISSUE TITLE
TypeError in 'timewarp --inspect --pull-exif --verbose'

### ISSUE THREAD

**Issue by aa599** [RhetTbull/osxphotos#976]
**Describe the bug**
Selected photos in a shared library, ran 'osxphotos timewarp --inspect --pull-exif --verbose', got

TypeError: '>=' not supported between instances of 'NoneType' and 'int'

It'd be helpful if the stacktrace mentioned which photo it was working on when the exception occurred.

**To Reproduce**
2. What was the error output?

```
Proceed with timewarp? [y/N]: y
filename, uuid, photo time (local), photo time, timezone offset, timezone name
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Library/Frameworks/Python.framework/Versions/3.8/bin/osxphotos:8 in <module>                    │
│                                                                                                  │
│   5 from osxphotos.__main__ import cli_main                                                      │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 8 │   sys.exit(cli_main())                                                                     │
│   9                                                                                              │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py:1130 │
│ in __call__                                                                                      │
│                                                                                                  │
│   1127 │                                                                                         │
│   1128 │   def __call__(self, *args: t.Any, **kwargs: t.Any) -> t.Any:                           │
│   1129 │   │   """Alias for :meth:`main`."""                                                     │
│ ❱ 1130 │   │   return self.main(*args, **kwargs)                                                 │
│   1131                                                                                           │
│   1132                                                                                           │
│   1133 class Command(BaseCommand):                                                               │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py:1055 │
│ in main                                                                                          │
│                                                                                                  │
│   1052 │   │   try:                                                                              │
│   1053 │   │   │   try:                                                                          │
│   1054 │   │   │   │   with self.make_context(prog_name, args, **extra) as ctx:                  │
│ ❱ 1055 │   │   │   │   │   rv = self.invoke(ctx)                                                 │
│   1056 │   │   │   │   │   if not standalone_mode:                                               │
│   1057 │   │   │   │   │   │   return rv                                                         │
│   1058 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!                               │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py:1657 │
│ in invoke                                                                                        │
│                                                                                                  │
│   1654 │   │   │   │   super().invoke(ctx)                                                       │
│   1655 │   │   │   │   sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)                    │
│   1656 │   │   │   │   with sub_ctx:                                                             │
│ ❱ 1657 │   │   │   │   │   return _process_result(sub_ctx.command.invoke(sub_ctx))               │
│   1658 │   │                                                                                     │
│   1659 │   │   # In chain mode we create the contexts step by step, but after the                │
│   1660 │   │   # base command has been invoked.  Because at that point we do not                 │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py:1404 │
│ in invoke                                                                                        │
│                                                                                                  │
│   1401 │   │   │   echo(style(message, fg="red"), err=True)                                      │
│   1402 │   │                                                                                     │
│   1403 │   │   if self.callback is not None:                                                     │
│ ❱ 1404 │   │   │   return ctx.invoke(self.callback, **ctx.params)                                │
│   1405 │                                                                                         │
│   1406 │   def shell_complete(self, ctx: Context, incomplete: str) -> t.List["CompletionItem"]:  │
│   1407 │   │   """Return a list of completions for the incomplete value. Looks                   │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py:760  │
│ in invoke                                                                                        │
│                                                                                                  │
│    757 │   │                                                                                     │
│    758 │   │   with augment_usage_errors(__self):                                                │
│    759 │   │   │   with ctx:                                                                     │
│ ❱  760 │   │   │   │   return __callback(*args, **kwargs)                                        │
│    761 │                                                                                         │
│    762 │   def forward(                                                                          │
│    763 │   │   __self, __cmd: "Command", *args: t.Any, **kwargs: t.Any  # noqa: B902             │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/osxphotos/cli/time │
│ warp.py:585 in timewarp                                                                          │
│                                                                                                  │
│   582 │   │   │   │   "[filename]filename[/filename], [uuid]uuid[/uuid], [time]photo time (loc   │
│   583 │   │   │   )                                                                              │
│   584 │   │   for photo in photos:                                                               │
│ ❱ 585 │   │   │   tz_seconds, tz_str, tz_name = tzinfo.get_timezone(photo)                       │
│   586 │   │   │   photo_date_local = datetime_naive_to_local(photo.date)                         │
│   587 │   │   │   photo_date_tz = datetime_to_new_tz(photo_date_local, tz_seconds)               │
│   588 │   │   │   rich_echo(                                                                     │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/osxphotos/phototz. │
│ py:69 in get_timezone                                                                            │
│                                                                                                  │
│    66 │   │   │   c.execute(sql)                                                                 │
│    67 │   │   │   results = c.fetchone()                                                         │
│    68 │   │   tz, tzname = (results[0], results[1])                                              │
│ ❱  69 │   │   tz_str = tz_to_str(tz)                                                             │
│    70 │   │   return tz, tz_str, tzname                                                          │
│    71                                                                                            │
│    72                                                                                            │
│                                                                                                  │
│ /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/osxphotos/phototz. │
│ py:21 in tz_to_str                                                                               │
│                                                                                                  │
│    18                                                                                            │
│    19 def tz_to_str(tz_seconds: int) -> str:                                                     │
│    20 │   """convert timezone offset in seconds to string in form +00:00 (as offset from GMT)"   │
│ ❱  21 │   sign = "+" if tz_seconds >= 0 else "-"                                                 │
│    22 │   tz_seconds = abs(tz_seconds)                                                           │
│    23 │   # get min and seconds first                                                            │
│    24 │   mm, _ = divmod(tz_seconds, 60)                                                         │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: '>=' not supported between instances of 'NoneType' and 'int'

```

**Comment by RhetTbull** [comment]
@all-contributors please add @aa599 for bug

**Comment by RhetTbull** [comment]
Thanks for the traceback. This is caused by a photo in the database having no timezone value set. I've never seen this myself so hadn't put in handling for this case. The code will now check for this and set timezone to 0 if there is none set in the database. (GMT). As you suggested, the code will also now include the UUID and the filename of the photo currently being processed in the crash report.

**Comment by RhetTbull** [comment]
Fixed in [v0.57.1](https://github.com/RhetTbull/osxphotos/releases/tag/v0.57.1)
